### ISSUE TITLE
Comments in lists are duplicated during file merges

### ISSUE THREAD

**Issue by bruce-au** [mikefarah/yq#730]
**Describe the bug**
When merging two files, we see comments in lists getting duplicated.
Version of yq: 4.6.0
Operating system: Mac
Installed via: homebrew

**Input Yaml**

data1.yml:
```yaml
list:
# Hi this is a comment.
# Hello this is another comment.
- "abc"
```

data2.yml:
```yaml
list2:
# This is yet another comment.
# Indeed this is yet another comment.
- "123"
```

**Command**
The command you ran:
```
yq eval-all 'select(fi == 0) * select(fi == 1)' data1.yaml data2.yaml
```

**Actual behavior**

```yaml
list:
  # Hi this is a comment.
  # Hello this is another comment.# Hi this is a comment.
  # Hello this is another comment.
  - "abc"
list2:
  # This is yet another comment.
  # Indeed this is yet another comment.# This is yet another comment.
  # Indeed this is yet another comment.
  - "123"
 ```

**Expected behavior**

```yaml
list:
  # Hi this is a comment.
  # Hello this is another comment.
  - "abc"
list2:
  # This is yet another comment.
  # Indeed this is yet another comment.
  - "123"
```

**Comment by mikefarah** [comment]
Verified bug - thanks for the clear issue

**Comment by mikefarah** [comment]
Fixed in 4.6.2
